verify/update-bazel.sh

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env bash
+# Copyright 2016 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+REPOINFRA_ROOT=$(git rev-parse --show-toplevel)
+# https://github.com/kubernetes/test-infra/issues/5699#issuecomment-348350792
+cd ${REPOINFRA_ROOT}
+TMP_GOPATH=$(mktemp -d)
+
+# no unit tests in vendor
+# previously we used godeps which did this, but `dep` does not handle this
+# properly yet. some of these tests don't build well. see:
+# ref: https://github.com/kubernetes/test-infra/pull/5411
+find ${REPOINFRA_ROOT}/vendor/ -name "*_test.go" -delete
+
+# manually remove BUILD file for github.com/bazelbuild/buildtools/BUILD.bazel if it
+# exists; there is a specific test_suite rule that breaks importing
+rm -f ${REPOINFRA_ROOT}/vendor/github.com/bazelbuild/buildtools/BUILD.bazel
```

Comment:
ugh, mixing go vendoring + bazel is the worst. :(

Reply:
I wonder if it's worth just.. find/deleting all BUILD/BUILD.bazel files in `vendor/`. @BenTheElder had suggested this before in test-infra, but I wasn't sure if there was possibly other rules in those files that would be useful in the context of go vendoring (when we strip tests and other files).

Reply:
This SGTM except that it would stink to need to regenerate *all* of these *every* update run, lots of update runs shouldn't touch `vendor/` at all 🤔

Reply:
@BenTheElder We could perhaps use `kube::util::has_changes_against_upstream_branch` to check if there has been changes in `vendor/` against upstream, and only trigger if there has. This is used for verifying core godeps only if there has been changes (or forced in CI)

Reply:
Deleting all existing `BUILD` files can be dangerous, since there might be `genrule`s or other customizations we need.

I filed https://github.com/bazelbuild/bazel-gazelle/issues/63 upstream to figure out a better long-term fix for this issue.